src/map/battle.c

```diff
@@ -4359,8 +4359,9 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 	int i, nk;
 	bool n_ele = false; // non-elemental
 
-	struct map_session_data *sd, *tsd;
 	struct Damage wd;
+	struct map_session_data *sd = BL_CAST(BL_PC, src);
```

Comment:
Both `sd` and `tsd` are assigned few lines below, please move the declaration there if you're changing this.